community_server/src/Controller/AppRequestsController.php

```diff
@@ -265,6 +266,89 @@ public function createCoins()
         
     }
     
+    public function getBalance($session_id = 0)
+    {
+
+        $login_result = $this->requestLogin($session_id, false);
+        if($login_result !== true) {
+            return $this->returnJson($login_result);
+        }
+        $session = $this->getRequest()->getSession();
+        $user = $session->read('StateUser');
+        $state_balances_table = TableRegistry::getTableLocator()->get('StateBalances');
+        $state_balances_table->updateBalances($user['id']);
+        
+        $state_balance = $state_balances_table->find()->where(['state_user_id' => $user['id']])->first();
+
+        if(!$state_balance) {
+            return $this->returnJson(['state' => 'success', 'balance' => 0]);
+        }
+        $now = new FrozenTime();
+        return $this->returnJson([
+            'state' => 'success',
+            'balance' => $state_balance->amount,
+            'decay' => $state_balance->partDecay($now),
+            'decay_date' => $now
+        ]);
+    }
+    
+    public function listTransactions($page = 1, $count = 25, $orderDirection = 'ASC', $session_id = 0)
+    {
+        $startTime = microtime(true);
+        $login_result = $this->requestLogin($session_id, false);
+        if($login_result !== true) {
+            return $this->returnJson($login_result);
+        }
+        $session = $this->getRequest()->getSession();
+        $user = $session->read('StateUser');
+        
+        
+        $stateBalancesTable = TableRegistry::getTableLocator()->get('StateBalances');
+        $stateUserTransactionsTable = TableRegistry::getTableLocator()->get('StateUserTransactions');
+        $transactionsTable  = TableRegistry::getTableLocator()->get('Transactions');
+        
+        
+        $stateBalancesTable->updateBalances($user['id']);
+
+        $gdtSum = 0;        
+        $gdtEntries = $this->JsonRequestClient->sendRequestGDT(['email' => $user['email']], 'GdtEntries' . DS . 'sumPerEmailApi');
+
+        if('success' == $gdtEntries['state'] && 'success' == $gdtEntries['data']['state']) {
+          $gdtSum = intval($gdtEntries['data']['sum']);
+        } else {
+          if($user) {   
+            $this->addAdminError('StateBalancesController', 'overview', $gdtEntries, $user['id']);
+          } else {
+            $this->addAdminError('StateBalancesController', 'overview', $gdtEntries, 0);
```

Comment:
```suggestion
          $this->addAdminError('StateBalancesController', 'overview', $gdtEntries, $user['id'] ? $user['id'] : 0);
```

Can we also have the api calls for those documented pls ❤️ ?

Reply:
For what do you mean?
addAdminError isn't a API Call. 
But I can make this accessable via API.

Reply:
The suggestion shortens the code quite a bit. Not sure about performance - if there is a difference if you use a ternary operator vs if else.
Yes an API would be quite fancy to have